### PR TITLE
V6 theme re org

### PR DESCRIPTION
**Pull request by colinlohner**
## Overview
New organization of theming. Default styles are now created using a method so that they are theme aware (currently only applied to buttons). Passing custom styles and custom variants is still allowed/available and now token overwrites work as expected.

Passing custom variant:
```
// Theme object
const customTheme = {
  componentName: {
    variantName: {
      base: {
        backgroundColor: 'red'
      }
    }
  }
}

<ThemeProvider value={customTheme}>
```

Updating token
```
// Theme object
const customTheme = {
  tokens: {
    primary: {
      base: 'green'
    }
  }
}

<ThemeProvider value={customTheme}>
```

## Screenshots (if applicable)
n/a 

## Testing

- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/theme/src/default-styles/buttons.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { getDefaultControlStyles } from './shared'
```
Comment (akleiner2) [inline src/theme/src/default-styles/buttons.js:1]: You're essentially lumping this as the new `default-theme`?

**Comment by akleiner2** [comment]
Cool! I think this looks good 👍 . Can't wait till we're able to get rid of the other cruft here too

**Comment by akleiner2** [comment]
Btw - for disabled buttons, can we add a `cursor: not-allowed`? In addition to just making the background `<color>100`

**Comment by colinlohner** [comment]
@akleiner2 that's actually there, coming from `shared.js` disabled styles

**Comment by colinlohner** [comment]
may not be visible though, due to the `pointer-events: none`

**Pull request merged**